emacs/check.sh

```diff
@@ -3,12 +3,12 @@
 # Adapted from https://github.com/purcell/package-lint/blob/master/run-tests.sh
 EMACS="${EMACS:=emacs}"
 
-NEEDED_PACKAGES="package-lint company iedit auto-complete"
+NEEDED_PACKAGES="package-lint company iedit auto-complete compat"
 
 ELS_TO_CHECK=*.el
 # To reduce the amount of false positives we only package-lint files
 # that are actual installable packages.
-PKGS_TO_CHECK="merlin.el merlin-ac.el merlin-company.el merlin-iedit.el"
+PKGS_TO_CHECK="merlin.el merlin-ac.el merlin-company.el merlin-iedit.el merlin-cap.el tests/merlin-cap-test.el"
```

Comment:
I don't think this is right any more. From a packaging point of view, `merlin`, `merlin-ac`, `merlin-company` and `merlin-iedit` are separate packages, because their dependencies are different. Each is built into a separate MELPA package, for granular installation.

However, presumably the intention is that `merlin-cap` will be part of the base `merlin` package, since it doesn't depend on anything opinionated that ships separately from Emacs. So it wouldn't be appropriate to run `package-lint` on `merlin-cap` separately: instead, run `package-lint` on both `merlin-cap.el` and `merlin.el` together, with the `package-lint-main-file` variable set to `merlin.el`.

And `merlin-cap-test.el` doesn't need package-linting, because it's not shipped in the package.

Reply:
Thanks, I did that in [a72fd55](https://github.com/ocaml/merlin/pull/1827/commits/a72fd55988ba1c5a00915668e247238adc469295).

There a still spurious `merlin-cap.el: error: You should depend on (emacs "28.1") if you need `string-search'.` even if we use the compat module.


```
merlin.el:477:0: error: `merlin/unmake-point' contains a non-standard separator `/', use hyphens instead (see Elisp Coding Conventions).
merlin.el:608:0: error: `merlin/call' contains a non-standard separator `/', use hyphens instead (see Elisp Coding Conventions).
merlin.el:1194:0: error: `merlin/display-in-type-buffer' contains a non-standard separator `/', use hyphens instead (see Elisp Coding Conventions).
merlin-cap.el:10:0: error: Package-Requires outside the main file have no effect.
merlin-cap.el:341:13: error: You should depend on (emacs "28.1") if you need `string-search'.
merlin-cap.el:356:19: error: You should depend on (emacs "28.1") if you need `string-search'.
merlin-cap.el:376:18: error: You should depend on (emacs "28.1") if you need `string-search'.
```

Reply:
Ah, I was overdue to tag a new `package-lint` release, so this spurious message was due to the build picking up an old version. Once MELPA has rebuilt the new `package-lint` version this complaint should go away.